code/API_definitions/Edge-Application-Management.yaml

```diff
@@ -961,6 +999,278 @@ components:
+                             such as macvlan, ipvlan
+                  Note: The use of SR-IOV interfaces automatically
+                  configure the required kernel parameters for the nodes.
+                type: string
+                example: vfio-pci
+                enum:
+                  - netdevice
+                  - vfio-pci
+                  - interface
+
+    AdditionalStorage:
+      description: Additional storage for the application.
+      type: array
+      items:
+        type: object
+        required:
+          - storageSize
+          - mountPoint
+        properties:
+          name:
+            type: string
+            description: Name of additional storage resource.
+            example: logs
+          storageSize:
+            type: string
+            description: Additional persistent volume for the application.
+            example: 80GB
+            pattern: ^\d+(GB|MB)$
+          mountPoint:
+            type: string
+            description: Location of additional storage resource.
+            example: /logs
+
+    Vcpu:
+      type: string
+      pattern: ^\d+((\.\d{1,3})|(m))?$
+      description: |
+        Number of vcpus in whole (i.e 1), decimal (i.e 0.500) up to
+        millivcpu, or millivcpu (i.e 500m) format.
+      example: "500m"
+
+    KubernetesResources:
+      description: Definition of Kubernetes Cluster Infrastructure.
+      type: object
+      required:
+        - nodePools
+        - infraKind
+        - applicationResources
+      properties:
+        infraKind:
+          description: Type of infrastructure for the application.
+          type: string
+          example: kubernetes
+          enum:
+            - kubernetes
+        applicationResources:
+            $ref: '#/components/schemas/ApplicationResources'
+        version:
+          type: string
+          description: Minimum Kubernetes Version.
+          example: "1.29"
+        isShared:
+          type: boolean
+          description: |
+            True: Enable the operator platform to reuse the kubernetes cluster
+            False: Disable the operator platform to reuse the kubernetes cluster
+          default: false
+        nodePools:
+          type: array
+          description: |
+            Description of worker node set in a Kubernetes cluster.
+          items:
+            $ref: '#/components/schemas/NodePool'
+        additionalStorage:
+          type: string
+          description: |
+            Amount of persistent storage allocated to the Kubernetes PVC.
+          example: 80GB
+          pattern: ^\d+(GB|MB)$
+        networking:
+          $ref: '#/components/schemas/K8sNetworking'
+        addons:
+          $ref: '#/components/schemas/K8sAddons'
+
+    KubernetesResourcesRef:
+      type: object
+      description: Kubernetes reference object
+      required:
+        - infraKind
+        - kubernetesClusterRef
+        - applicationResources
```

Comment:
As my previous comment i suggest to delink applicationResources from KubernetesResourcesRef. We may keep KubernetesResourcesRef to associate kubernetes entities only and let other abstractions which needs it can refer to it.

Reply:
I agree, We will propose a change for that.

Reply:
The latest commit to this PullRequest ontroduces modifications to cover this and previous comments. **The concept of flavours has been removed** so that the cpu/memory required for an application is directly defined by the API consumer (the application porvider). 

For ths case of kubernetes, the case **where an application is expected to be deployed over a k8s cluster**, the concept of _resource topology_ is  introduced in order to allow the API consumer (the application prvider) describing what is the minimum k8s infrastructure  required to host the application. The topology describes the minimum number of nodes required and the minimum cpu/memory required per node. 

**Also,** in order to cope with applications with different components that need different hardware (e.g.: GPU required in some component) the concept of hardware pool has been introduced to allow th application provider to define the specific hardware configuration for each application component. **With this information, the API server (i.e.: an edge orchestrator) will be able to determine the requirements of the k8s cluster and nodepools needed, to host the application components**. In this proposal we start with 2 types of pools (CPU and GPU) that can be later extended to additional specific hardware (e.g.: future TPU or other specific processors to consider in the future).

**Finally**, to cope with scenarios of shared k8s clusters, the parameters _isStandalone_ and _kubernetesClusterRef_ have been added.